### PR TITLE
test: improve stability for tests

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #773

### What is changed and how it works?

dynamic increase waiting time (by retry) for
`testForEtcd.TestInfoEtcd`, `testForEtcd.TestOperationEtcd`, `testOptimist.testOptimist`, `sharding2`

increase lease TTL for
`testMockScheduler`

add notes for
`testMaster.TestOfflineMember`, `testForEtcd.TestWorkerKeepAlive`, `testMockScheduler`, `ha_cases`

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by lance6716** [comment]
WIP: change to utils.WaitSomething

**Comment by lance6716** [comment]
PTAL @WangXiangUSTC @GMHDBJD

File: pkg/shardddl/pessimism/info_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	. "github.com/pingcap/check"
+	"github.com/pingcap/dm/pkg/utils"
```
Comment (WangXiangUSTC) [inline pkg/shardddl/pessimism/info_test.go:24]: how about move it after L29

File: dm/master/shardddl/optimist_test.go [hunk 2]
```diff
@@ -254,8 +254,11 @@ func (t *testOptimist) testOptimist(c *C, restart int) {
 	// wait operation for i11 become available.
 	opCh := make(chan optimism.Operation, 10)
 	errCh := make(chan error, 10)
-	ctx2, cancel2 := context.WithTimeout(ctx, watchTimeout)
-	optimism.WatchOperationPut(ctx2, etcdTestCli, i11.Task, i11.Source, i11.UpSchema, i11.UpTable, rev1, opCh, errCh)
+	ctx2, cancel2 := context.WithCancel(ctx)
+	go optimism.WatchOperationPut(ctx2, etcdTestCli, i11.Task, i11.Source, i11.UpSchema, i11.UpTable, rev1, opCh, errCh)
+	utils.WaitSomething(10, watchTimeout, func() bool {
```
Comment (WangXiangUSTC) [inline dm/master/shardddl/optimist_test.go:259]: should use assert?

File: dm/master/shardddl/optimist_test.go [hunk 3]
```diff
@@ -309,8 +312,11 @@ func (t *testOptimist) testOptimist(c *C, restart int) {
 	// wait operation for i12 become available.
 	opCh = make(chan optimism.Operation, 10)
 	errCh = make(chan error, 10)
-	ctx2, cancel2 = context.WithTimeout(ctx, watchTimeout)
-	optimism.WatchOperationPut(ctx2, etcdTestCli, i12.Task, i12.Source, i12.UpSchema, i12.UpTable, rev2, opCh, errCh)
+	ctx2, cancel2 = context.WithCancel(ctx)
+	go optimism.WatchOperationPut(ctx2, etcdTestCli, i12.Task, i12.Source, i12.UpSchema, i12.UpTable, rev2, opCh, errCh)
+	utils.WaitSomething(10, watchTimeout, func() bool {
```
Comment (WangXiangUSTC) [inline dm/master/shardddl/optimist_test.go:317]: ditto

File: dm/master/shardddl/optimist_test.go [hunk 4]
```diff
@@ -356,8 +362,11 @@ func (t *testOptimist) testOptimist(c *C, restart int) {
 	// wait operation for i21 become available.
 	opCh = make(chan optimism.Operation, 10)
 	errCh = make(chan error, 10)
-	ctx2, cancel2 = context.WithTimeout(ctx, watchTimeout)
-	optimism.WatchOperationPut(ctx2, etcdTestCli, i21.Task, i21.Source, i21.UpSchema, i21.UpTable, rev1, opCh, errCh)
+	ctx2, cancel2 = context.WithCancel(ctx)
+	go optimism.WatchOperationPut(ctx2, etcdTestCli, i21.Task, i21.Source, i21.UpSchema, i21.UpTable, rev1, opCh, errCh)
+	utils.WaitSomething(10, watchTimeout, func() bool {
```
Comment (WangXiangUSTC) [inline dm/master/shardddl/optimist_test.go:367]: ditto

File: pkg/shardddl/pessimism/operation_test.go [hunk 5]
```diff
@@ -65,8 +66,12 @@ func (t *testForEtcd) TestOperationEtcd(c *C) {
 	// start the watcher with the same revision as the last PUT for the specified task and source.
 	wch := make(chan Operation, 10)
 	ech := make(chan error, 10)
-	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
-	WatchOperationPut(ctx, etcdTestCli, task1, source1, rev2, wch, ech)
+	ctx, cancel := context.WithCancel(context.Background())
+	go WatchOperationPut(ctx, etcdTestCli, task1, source1, rev2, wch, ech)
+	// wait response of WatchOperationPut, increase waiting time when resource shortage
+	utils.WaitSomething(10, 500*time.Millisecond, func() bool {
```
Comment (WangXiangUSTC) [inline pkg/shardddl/pessimism/operation_test.go:72]: ditto

File: dm/master/shardddl/optimist_test.go (hunk 2 above)
Comment (lance6716) [reply dm/master/shardddl/optimist_test.go:259]: assert is in lines below https://github.com/pingcap/dm/pull/785/files#diff-2bb523adf53949e58e838fc0685db092R265 , github doesn't display it.

these code just dynamic increase waiting time, for best case, it waits `watchTimeout ` which is same as original code, else it trys to wait more. after timeout, assertion is called.

File: dm/master/shardddl/optimist_test.go (hunk 3 above)
Comment (lance6716) [reply dm/master/shardddl/optimist_test.go:317]: same

File: dm/master/shardddl/optimist_test.go (hunk 4 above)
Comment (lance6716) [reply dm/master/shardddl/optimist_test.go:367]: same

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**